sympy/physics/control/control_plots.py

```diff
@@ -0,0 +1,160 @@
+
+matplotlib = import_module(
+        'matplotlib', import_kwargs={'fromlist': ['pyplot']},
+        catch=(RuntimeError,))
+
+numpy = import_module('numpy')
+
+if matplotlib:
+    plt = matplotlib.pyplot
+
+if numpy:
+    np = numpy  # Matplotlib already has numpy as a compulsory dependency. No need to install it separately.
+
+
+def pole_zero(system, pole_colour='r', zero_colour='b', grid=True, show=True, **kwargs):
+    r"""
+    Returns the Pole-Zero plot (also known as PZ Plot or PZ Map) of a system.
+    """
+    system = system.doit()  # Get the equivalent TransferFunction object.
+
+    num_poly = Poly(system.num, system.var).all_coeffs()
+    den_poly = Poly(system.den, system.var).all_coeffs()
+
+    num_poly = np.array(num_poly, dtype=np.float64)
+    den_poly = np.array(den_poly, dtype=np.float64)
+
+    zeros, poles = _find_roots(num_poly, den_poly)
+
+    zero_real = np.real(zeros)
+    zero_imag = np.imag(zeros)
+
+    pole_real = np.real(poles)
+    pole_imag = np.imag(poles)
+
+    plt.plot(pole_real, pole_imag, 'x', mfc='none', markersize=10)
+    plt.plot(zero_real, zero_imag, 'o', markersize=7)
+    plt.xlabel('Real')
+    plt.ylabel('Imaginary')
+    plt.title('Poles and Zeros')
+    if grid:
+        plt.grid()
+    plt.axhline(0, color='black')
+    plt.axvline(0, color='black')
+    if show:
+        plt.show()
+        return
+    return plt
+
+
+def step_response(system, show_input=False, colour='r', show=True, grid=True, upper_limit=6, **kwargs):
+    r"""
+    Return the unit step response of a continuous-time system.
+    """
+    x = Symbol("x")
+    expr = system.to_expr()/(system.var)
+    y = inverse_laplace_transform(expr, system.var, x)
+    return plot(y, (x, 0, upper_limit), show=show, title="Unit Step Response", \
+        xlabel="Time (s)", ylabel="Amplitude")
+
+
+def impulse_response(system, show_input=False, colour='r', show=True, grid=True, upper_limit=6, **kwargs):
+    r"""
+    Return the unit impulse response (Input is the Dirac-Delta Function) of a
+    continuous-time system.
+    """
+    x = Symbol("x")
+    expr = system.to_expr()
+    y = inverse_laplace_transform(expr, system.var, x)
+    return plot(y, (x, 0, 6), show=True, title="Impulse Response", \
+        xlabel="Time (s)", ylabel="Amplitude")
+
+
+def ramp_response(system, slope=1, show_input=False, colour='r', show=True, grid=True, upper_limit=6, **kwargs):
+    r"""
+    Return the unit impulse response (Input is the Dirac-Delta Function) of a
+    continuous-time system.
+    """
+    x = Symbol("x")
+    expr = (slope*system.to_expr())/((system.var)**2)
+    y = inverse_laplace_transform(expr, system.var, x)
+    return plot(y, (x, 0, upper_limit), show=show, title="Ramp Response", \
+        xlabel="Time (s)", ylabel="Amplitude")
+
+
+def bode_plot(system, initial_exp=-5, final_exp=5, show=True, **kwargs):
+    r"""
+    Plot a Bode plot of a continuous-time system.
+    """
+    expr = system.to_expr()
```

Comment:
Check the arguments of the plotting functions?

Reply:
As told earlier, the user API is crude now. I have not dealt with input checking yet. Even the keyword arguments passed are unused in most of the function signatures.